tutorials/04 - DistortionEffectDemo.md

```diff
@@ -0,0 +1,138 @@
+# 04 - Distortion Effect
+
+In this tutorial, we're going to analyze the structure of an audio effect application, by applying distortion to a looping audio file.
+
+## Main Tracktion Engine Classes
+#### Engine
+Every tracktion Engine app starts with an 'Engine' instance. This performs the initialisation and shutdown of the engine.
+
+#### Edit
+The 'Edit' is the container for a playable arrangement. it holds tracks, tempo sequences, Racks etc.
+
+#### Track
+'Edit' contains a list of 'Track's. This project makes use of the 'AudioTrack' which outputs audio.
+
+#### Clip
+'Clip' is the base class for clips that live in 'Track's. 'Clip's can be MIDI, audio, step or the edit-in-an-edit 'editClip.'
+
+
+## Main JUCE Classes
+#### Plugin
+
+## DistortionEffectDemo.h
+
+
+## DistortionPlugin Class
```

Comment:
I'd add an into as to what the DistortionPlugin class is here. I.e. a Plugin subclass so it can be added as a plugin to clips/tracks etc. in an Edit. For demo purposes we're using just a very simple tanh waveshaper